src/com/sun/ts/tests/websocket/spec/servercontainer/upgradehttptowebsocket/WSTestServlet.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package com.sun.ts.tests.websocket.spec.servercontainer.upgradehttptowebsocket;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.websocket.DeploymentException;
+import jakarta.websocket.server.ServerContainer;
+import jakarta.websocket.server.ServerEndpointConfig;
+
+@WebServlet("/TCKTestServlet")
+public class WSTestServlet extends HttpServlet {
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    ServerContainer sc = (ServerContainer) getServletContext().getAttribute("jakarta.websocket.server.ServerContainer");
+    ServerEndpointConfig sec = ServerEndpointConfig.Builder.create(WSTestServer.class, "/TCKTestServer").build();
+    try {
+      sc.upgradeHttpToWebSocket(req, resp, sec, Collections.emptyMap());
```

Comment:
The tests under `spec` folder usually check the assertions required by the Spec document (pdf). This test seems to check this new API method. Should it be in the `ee` test folder then?

Reply:
Just for quick reference, [the websocket test folder contains subfolders:](https://github.com/eclipse-ee4j/jakartaee-tck/tree/master/src/com/sun/ts/tests/websocket):
```
- api
- common
- ee
- negdep
- platform
- spec
```

I'm not sure about the purpose of the `spec` folder myself versus `ee` but typically, I thought `ee` is for the `Jakarta EE Full Platform` + `Web Profile` specific tests.  

@LanceAndersen do you know the answer to [Jan's question in the comment above](https://github.com/eclipse-ee4j/jakartaee-tck/pull/783#discussion_r786868348)?

Having a README.me file in the websocket folder that mentions what each folder is for could be helpful.

Reply:
It depends on the TCK. Some TCK uses ee for end-to-end tests (WebSocket+JAX-RS), some use ee for enterprise tests (I think JAX-WS, JPA) as opposed to SE tests. Platform folder contains tests for which a full platform is required, for instance for interoperability with CDI, i.e. tests that are only full CTS tests, and are not part of the standalone TCKs.

Reply:
> It depends on the TCK. Some TCK uses ee for end-to-end tests (WebSocket+JAX-RS), some use ee for enterprise tests (I think JAX-WS, JPA) as opposed to SE tests. Platform folder contains tests for which a full platform is required, for instance for interoperability with CDI, i.e. tests that are only full CTS tests, and are not part of the standalone TCKs.

Thanks @jansupol, I had no idea of ^.  If there are rules about which folders can include different types of tests, those rules should be documented somewhere, otherwise the pull request reviewers and contributors will have no idea (like me :-).  

Personally, I don't yet fully understand the purpose of each of the above mentioned folders.  Perhaps `api` should instead be called `javadoc` (it seems to only have assertions for javadoc).  In the `ee` (end-to-end) folder, I see both `javadoc` + `spec` tests (same in other folders as well).  

This PR is now blocking the Platform TCK from building (we are getting build failures) which means we cannot run Platform TCK testing to see where we are.  IMO, this needs to be solved so that EE 10 Platform implementations can test if they want to.

CC @gurunrao @alwin-joseph

Reply:
I assume you mean you need to merge this PR to fix the platform TCK build. If that is the case I'd suggest:
- merge this PR now
- if there is an issue with the locations of the tests I've added open an issue and assign it to me
- I'll provide a PR to put the new tests in the right place